workspaces/diff-engine/src/events/macros.rs

```diff
@@ -0,0 +1,124 @@
+/// Compactly implement Events or groups of Events. Implements `Event` and `WithEventContext` traits and links between groups.
+///
+/// # Examples
+///
+/// The most basic use is a single event struct, where one provides the struct and it's &'static str event name
+/// ```
+/// # use optic_diff_engine::EventContext;
+/// # use optic_diff_engine::cqrs_event;
+/// #
+/// struct ExampleEvent {
+///   some_property: String,
+///   // struct is required to have an optional event_context
+///   event_context: Option<EventContext>,
+/// }
+///
+/// cqrs_event!(ExampleEvent => "example_event_name");
+/// ```
```

Comment:
I tried to make sure the macro is well documented, especially because their implementation is harder to read for those who haven't worked with them yet. By doing that, I learned that cargo compiles and runs all code blocks as tests as part of `cargo test`, making sure they stay up-to-date. Neat-o!